manager/dispatcher/dispatcher.go

```diff
@@ -759,6 +759,7 @@ func (d *Dispatcher) Assignments(r *api.AssignmentsRequest, stream api.Dispatche
 		initial   api.AssignmentsMessage
 	)
 	tasksMap := make(map[string]*api.Task)
+	tasksUsingSecret := make(map[string]map[string]struct{})
```

Comment:
I can refactor a bit to try to make it less complex, but would love to not hold up the PR on it, if possible.